homeassistant/components/modbus/binary_sensor.py

```diff
@@ -46,13 +56,18 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 class ModbusCoilSensor(BinarySensorDevice):
     """Modbus coil sensor."""
 
-    def __init__(self, hub, name, slave, coil):
+    def __init__(self, hub, name, slave, coil, register_type):
         """Initialize the Modbus coil sensor."""
         self._hub = hub
         self._name = name
         self._slave = int(slave) if slave else None
         self._coil = int(coil)
         self._value = None
+        self._register_type = (
+            REGISTER_TYPE_COIL
+            if register_type == REGISTER_TYPE_COIL
```

Comment:
WHy do you need the conditional checking provided register_type is always set - it has a default value and otherwise is validated by the voluptious schema. Just do the assignment directly.